spyderlib/widgets/file_switcher.py

```diff
@@ -0,0 +1,405 @@
+# -*- coding: utf-8 -*-
+#
+# Augmented file switcher by DM, 2015 (based on original in widgets/editor.py)
+# Licensed under the terms of the MIT License
+# (see spyderlib/__init__.py for details)
+
+from __future__ import print_function
+
+from spyderlib.baseconfig import _
+from spyderlib.qt.QtGui import (QVBoxLayout, QHBoxLayout,
+                                QListWidget, QListWidgetItem,
+                                QDialog, QLineEdit, QKeyEvent,
+                                QStyledItemDelegate, QStyleOptionViewItemV4, 
+                                QApplication, QTextDocument, QStyle,
+                                QAbstractTextDocumentLayout, QToolButton,
+                                QToolTip)
+from spyderlib.qt.QtCore import (Signal, Qt,  QObject,
+                                  QSize, QEvent, QPoint)
+
+import os
+import os.path as osp
+
+# Local imports
+from spyderlib.guiconfig import  new_shortcut
+from spyderlib.py3compat import to_text_string
+from spyderlib.utils.qthelpers import get_std_icon
+
+class HTMLDelegate(QStyledItemDelegate):
+    """ Taken from http://stackoverflow.com/a/5443112/2399799
+    """
```

Comment:
I think we need a better explanation of what this is doing, and then the reference to the place more like:

```python

class HTMLDelegate(QStyledItemDelegate):
    """Some nice one line explanation.

    Some more lengthy explanation if needed (optional)    

    Inspired by http://stackoverflow.com/a/5443112/2399799
    """